packages/fast-components/src/anchored-region/anchored-region.ts

```diff
@@ -0,0 +1,978 @@
+        this.disconnectAnchor(oldvalue);
+        if (this.initialLayoutComplete) {
+            this.initialLayoutComplete = false;
+            this.reset();
+        }
+    }
+
+    public viewportElement: HTMLElement | null;
+    private viewportElementChanged(oldvalue, newvalue): void {
+        this.disconnectViewport(oldvalue);
+        if (this.initialLayoutComplete) {
+            this.initialLayoutComplete = false;
+            this.reset();
+        }
+    }
+
+    /**
+     * values to be applied to the component's transform origin attribute on render
+     */
+    private transformOrigin: string;
+
+    /**
+     * values to be applied to the component's positioning attributes on render
+     */
+    private regionTop: string;
+    private regionRight: string;
+    private regionBottom: string;
+    private regionLeft: string;
+
+    /**
+     * the span in pixels of the selected position on each axis
+     */
+    private regionWidth: string;
+    private regionHeight: string;
+
+    private xTransformOrigin: string;
+    private yTransformOrigin: string;
+
+    private collisionDetector: IntersectionObserver;
+
+    private viewportRect: ClientRect | DOMRect | null;
+    private positionerDimension: Dimension;
+
+    private anchorTop: number;
+    private anchorRight: number;
+    private anchorBottom: number;
+    private anchorLeft: number;
+    private anchorHeight: number;
+    private anchorWidth: number;
+
+    private viewportScrollTop: number;
+    private viewportScrollLeft: number;
+
+    /**
+     * the positions currently being applied to layout
+     */
+    currentVerticalPosition: AnchoredRegionVerticalPositionLabel;
+    currentHorizontalPosition: AnchoredRegionHorizontalPositionLabel;
+
+    /**
+     * base offsets between the positioner's base position and the anchor's
+     */
+    private baseHorizontalOffset: number;
+    private baseVerticalOffset: number;
+
+    /**
+     * reference to the actual anchored container
+     */
+    public region: HTMLDivElement;
+
+    private openRequestAnimationFrame: number | null = null;
+
+    constructor() {
+        super();
+        this.handleAnchorResize = rafThrottle(this.handleAnchorResize);
+        this.handleRegionResize = rafThrottle(this.handleRegionResize);
+        this.setInitialState();
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+
+        if (isNil(this.viewportElement)) {
+            this.viewportElement = this.getViewport();
+        }
+
+        if (isNil(this.anchorElement)) {
+            this.anchorElement = this.getAnchor();
+        }
+
+        if (isNil(this.anchorElement) || isNil(this.viewportElement)) {
+            return;
+        }
+
+        this.connectObservers();
+
+        this.updateLayout();
+    }
+
+    public disconnectedCallback(): void {
+        super.disconnectedCallback();
+        this.disconnectObservers();
+    }
+
+    adoptedCallback() {
```

Comment:
I dunno, *do* you?

Reply:
The "to do" items are why it's still draft.  I believe we should do a reset but need to revisit to make sure.

Reply:
Added a reset.